salt/tests/unit/modules/test_metalk8s_package_manager_yum.py

```diff
@@ -0,0 +1,196 @@
+        """
+        with ForceImportErrorOn("yum"):
+            reload(metalk8s_package_manager_yum)
+            self.assertEqual(
+                metalk8s_package_manager_yum.__virtual__(),
+                False
+            )
+
+    @parameterized.expand(
+        param.explicit(kwargs=test_case)
+        for test_case in yaml.safe_load(open(YAML_TESTS_FILE))["list_dependents"]
+    )
+    def test_list_dependents(self, repoquery_output, result, **kwargs):
+        """
+        Tests the return of `_list_dependents` function
+        """
+        if repoquery_output is None:
+            repoquery_cmd_kwargs = {
+                'retcode': 1,
+                'stderr': 'An error has occured'
+            }
+        else:
+            repoquery_cmd_kwargs = {
+                'stdout': repoquery_output
+            }
+
+        repoquery_cmd_mock = MagicMock(
+            return_value=utils.cmd_output(**repoquery_cmd_kwargs)
+        )
+
+        with patch.dict(metalk8s_package_manager_yum.__salt__,
+                        {'cmd.run_all': repoquery_cmd_mock}):
+            self.assertEqual(
+                metalk8s_package_manager_yum._list_dependents(
+                    "my_package",
+                    "1.2.3",
+                    **kwargs
+                ),
+                result
+            )
+            repoquery_cmd_mock.assert_called_once()
+            self.assertEqual(
+                repoquery_cmd_mock.call_args[0][0][0],
+                "repoquery"
+            )
+            if "fromrepo" in kwargs:
+                self.assertTrue(
+                    kwargs["fromrepo"] in repoquery_cmd_mock.call_args[0][0]
+                )
+
+    @parameterized.expand(
+        param.explicit(kwargs=test_case)
+        for test_case in yaml.safe_load(open(YAML_TESTS_FILE))["list_pkg_dependents"]
+    )
+    def test_list_pkg_dependents(self, result,
+                                 list_dependents=None, rpm_qa_outputs=None,
+                                 **kwargs):
+        """
+        Tests the return of `list_pkg_dependents` function
+        """
+        def _rpm_qa_cmd(command):
+            out_kwargs = {}
+            if command[0] == 'rpm' and command[1] == '-qa':
+                # rpm_qa_outputs == None means nothings installed so
+                # `rpm -qa <package>` return "" and retcode 0
+                if rpm_qa_outputs is not None:
+                    if isinstance(rpm_qa_outputs, dict):
+                        if rpm_qa_outputs.get(command[2]) is None:
+                            out_kwargs['retcode'] = 1
+                            out_kwargs['stderr'] = 'An error has occured'
+                        else:
+                            out_kwargs['stdout'] = rpm_qa_outputs[command[2]]
+                    else:
+                        out_kwargs['stdout'] = rpm_qa_outputs
+                return utils.cmd_output(**out_kwargs)
+            return None
```

Comment:
I had a hard time understanding all the cases coming from different inputs to this test function. A test shouldn't have too much logic in it, so it's clear where the error comes from when the test breaks. Can't we split test cases, at least between nominal and error cases?

Reply:
This function is complicated? 
This function do not test anything about our product is just to simulate `rpm -qa` output from yaml file so that you can have different output depending on the package. It's just a helper to test

Reply:
I'll try to explain my point later when implementing tests for drain. I understand how this helper is just here to build a parametrized mock of `rpm -qa`, but it took me some time (maybe because there's no docstring, maybe because the test case which help me understand what `rpm_qa_outputs` look like, maybe because I'm dumb...).